src/Features/LanguageServer/Protocol/Handler/FoldingRanges/FoldingRangesHandler.cs

```diff
@@ -27,58 +25,13 @@ public FoldingRangesHandler()
 
         public async Task<FoldingRange[]> HandleRequestAsync(FoldingRangeParams request, RequestContext context, CancellationToken cancellationToken)
         {
-            var foldingRanges = ArrayBuilder<FoldingRange>.GetInstance();
```

Comment:
All existing code refactored as-is to a helper type that can be shared with LsifGenerator project.